### DISCUSSION TITLE
Does publishing a complete order JSON increase communication overhead in  Integration project example (version 6.1.2)?

### DISCUSSION THREAD

**Discussion by steel-tao** [openTCS/opentcs#175] [Q&A]
I deployed the Integration project example (version 6.1.2) and found in the test that when OperationDesk publishes an order in this version, it publishes the complete order JSON to the vehicle, while in the previous version (5.13.0), the order received by the vehicle usually only had two nodes and one edge. Is this a major change? Doesn't this increase the communication overhead?

**Comment by martingr** [comment]
The behaviour that you describe changed in version 0.13 of the openTCS VDA5050 communication adapter. (The current version of the integration project example uses version 0.22.) See the following excerpt from the [changelog](https://github.com/openTCS/opentcs-commadapter-vda5050/releases/tag/v0.13.0):

> A vehicle’s prospective route is included as the horizon in every order message. How many route steps are added to the horizon can be configured using the new vehicle property `vda5050:maxStepsHorizon`. For consistent naming, the existing property `vda5050:orderQueueSize` has been renamed to `vda5050:maxStepsBase`.

Please see the driver configuration document ([1] or [2], depending on the version you use) for information on the default values used with these properties. You can, of course, set `vda5050:maxStepsHorizon` to 0, which should correspond to the old behaviour.

[1]: https://github.com/openTCS/opentcs-commadapter-vda5050/blob/master/doc/v1.1/configuration.adoc
[2]: https://github.com/openTCS/opentcs-commadapter-vda5050/blob/master/doc/v2.0/configuration.adoc
